### PR TITLE
Major Refactoring for Input System

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Organized and moved quite a bit around.

- Removed a bunch of mapping classes we probably won't use
- Noticed we added pointer registration to the base input source, and I removed it again. That's not where we want to add that functionality but in a higher derived input source class that is specifically for spatial controllers that can have one or more or dynamically generated pointers.
- I'm going to try and make sure all our input sources are pure c#
- Refactored most to be except for InteractionInputSources and SpeechInputSource. SpeechInputSource will need a config profile to setup speech keywords and actions.
- Moved the controller visualization classes and helpers out of the input system too
- Ported a change to the SystemType to support structs

### CODE REVIEW

File: Assets/MixedRealityToolkit/InputSystem/Utilities/Enums/OculusControllerInputType.cs [hunk 1]
```diff
@@ -1,46 +0,0 @@
...
-
-using System;
-
-namespace Microsoft.MixedReality.Toolkit.InputSystem.Utilities
-{
-    [Flags]
-    public enum OculusControllerInputType
-    {
-        None                           = 0,
-        ButtonOnePress                 = 1 << 0,
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/InputSystem/Utilities/Enums/OculusControllerInputType.cs]: Maybe we should take some of these Enum Values for the Device Input Type (yay thought of a better name)

File: Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InputType.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InputType.cs]: Given the device nature of the use of this enum, I'll remove it from Input (or create a new one in _Core)

File: Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InteractionDefinition.cs [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        private Vector3 positionData;
+
+        /// <summary>
+        /// The rotation data storage for a 3DoF type.
+        /// </summary>
+        private Quaternion rotationData;
+
+        private Tuple<Vector3, Quaternion> transformData;
+
+        #endregion Definition Data items
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InteractionDefinition.cs]: you were supposed to leave in the single Get, but I'll re-add this after merge

File: Assets/MixedRealityToolkit/_Core/Interfaces/InputSystem/IMixedRealityInputAction.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem
```
Comment (SimonDarksideJ) [inline Assets/MixedRealityToolkit/_Core/Interfaces/InputSystem/IMixedRealityInputAction.cs]: Needs to be in "_Core" as it's consumed by BOTH device and Input

File: Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InteractionDefinition.cs (hunk 3 above)
Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/InteractionDefinition.cs]: Ah, my bad.

**Pull request merged**